### PR TITLE
Integration test for upgrading Ceph from mimic to nautilus

### PR DESCRIPTION
**Pull request by travisn**
Signed-off-by: travisn <tnielsen@redhat.com>

**Description of your changes:**
The integration test for upgrades will now upgrade from mimic to nautilus, in addition to testing the rook upgrade from the previous version of rook. 

This is dependent on the fix for #2973 which will allow the upgrade to succeed when the mons are updated to use msgr2.

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

### CODE REVIEW

File: tests/integration/upgrade_test.go [hunk 1]
```diff
@@ -147,6 +156,40 @@ func (s *UpgradeSuite) TestUpgradeToMaster() {
...
+
+	// Confirm that the upgrade was successful
+	// wait for the mds pods to be updated
+	// FIX: Use spec.CephVersionLabelKey
+	CephVersionLabelKey := "ceph-version"
+	expectedVersion := "14.2.0"
+	err = s.k8sh.WaitForDeploymentLabel(s.namespace, "rook-ceph-mon-a", CephVersionLabelKey, expectedVersion)
+	require.Nil(s.T(), err)
+
+	// wait for the rgw pods to be updated
```
Comment (leseb) [inline tests/integration/upgrade_test.go]:
```suggestion
	// wait for the osd pods to be updated
```

File: tests/integration/upgrade_test.go [hunk 2]
```diff
@@ -173,6 +216,22 @@ func (s *UpgradeSuite) updateClusterRoles() error {
 	}
 
 	newResources := `
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (BlaineEXE) [inline tests/integration/upgrade_test.go]: Does this need to be added to `upgrade-from-v0.9-create.yaml`?

Comment (travisn) [reply tests/integration/upgrade_test.go]: @BlaineEXE Yes, good catch, thanks

**Comment by travisn** [comment]
Rebasing and running the CI again, there sill seems to be some flaky aspects of the tests to work out...

**Comment by BlaineEXE** [comment]
Already implemented by another PR.

**Pull request closed without merging**